spinnaker-monitoring-daemon/tests/datadog_service_test.py

```diff
@@ -18,11 +18,49 @@
 import unittest
 import tempfile
 from mock import patch
-from StringIO import StringIO
 
 import datadog_service
-from datadog_service import DatadogMetricsService
-from ConfigParser import ConfigParser
+from datadog_service import DatadogMetricsService, DatadogArgumentsGenerator
+
+
+def service_generation_helper(config_data=[], datadog_options={},
+                              spinnaker_monitoring_options={}):
+    """
+    This utility lets you build a DatadogMetricsService in a DRY fashion
+    for all the tests below. You can inject additional parameters anywhere.
+
+    It guarantees the base invariants - api_key present,
+    'datadog' and 'dd_agent_config' keys exist in required arguments,
+    and that 'dd_agent_config' is correctly parsed as a file. These are
+    guarded against within the DatadogService.
+    """
+
+    data = ["[Main]", "api_key: FOUND_KEY"] + config_data
+    options = {'datadog': datadog_options, 'dd_agent_config': ''}
+    options.update(spinnaker_monitoring_options)
+    with tempfile.NamedTemporaryFile() as config:
```

Comment:
I think these files are leaked by the tests.

Do you think it practical to record these names in a list then have a TearDown method that deletes it? There are a number of ways you can do this though none obviously clean to me given you are calling this from multiple fixtures. The files are small so isnt threatening to leak them. It just seems unfortunate.

Reply:
I decided against implementing anything for this. Ideally, all tests should be run inside a Docker container anyway - mitigating leaked file descriptors for files with no sensitive information that are created only by tests that could be sandboxed did not seem worth investing time in compared to the net value received in doing it.

Reply:
I should also add I wasn't able to reproduce any leakage of file descriptors locally.